### PR TITLE
[FIX] host doc from PR on Circle-CI

### LINKED ISSUE #4401: Doc not hosted or deployed by circle CI
The last run happened 6 days ago: https://app.circleci.com/pipelines/github/nilearn/nilearn

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4401 

Changes proposed in this pull request:

- partly reverts https://github.com/nilearn/nilearn/pull/4402/files

### CODE REVIEW

File: .github/workflows/trigger_hosting_on_pr.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    # We get the url for the artifacts for circle-ci from https://nightly.link/
+    #
+    # This must be run in a separate workflow because for security reasons
+    # the secrets for login to circle-ci cannot be accessed by PRs opened from a fork.
+    #
+    # So this workflow waits for the doc building workflow to finish successfully
+    # before grabbing its output and pushing it to circle-CI.
+    #
+    # See https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+    #
```
Comment (Remi-Gau) [inline .github/workflows/trigger_hosting_on_pr.yml:12]:
```suggestion
    #
    # the CIRCLE_CI_TOKEN must be a personal token and not a project token
    # https://support.circleci.com/hc/en-us/articles/360060360811-CircleCI-API-v2-Returns-404-Not-Found
    #
```

**Comment by Remi-Gau** [comment]
pretty sure that this will not work until this is merged

**Review by bthirion — APPROVED** [review]
No obvious issue. I guess that the way forward is to merge-and-see.

**Comment by Remi-Gau** [comment]
FYI I tested on my fork that this would get triggered by a successful workflow on a pr.

Originally part of this workflow was in a bash script.

Maybe we can refactor after merging to make it easier to manage.

Same for documenting the workflow. Would do once we are sure it works.

**Comment by Remi-Gau** [comment]
@man-shu let me know if this makes sense to you

**Review by man-shu — APPROVED** [review]
Ok yes, let's merge it and then we can try this on some of the PRs

**Comment by Remi-Gau** [comment]
It worked. The PR that I updated has its doc on circle CI now.

Will update our internal Doc.

**Pull request merged**